jcache/remote/src/test/java/org/infinispan/jcache/TrackingCacheEntryListener.java

```diff
@@ -0,0 +1,95 @@
+package org.infinispan.jcache;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.concurrent.atomic.AtomicInteger;
+
+import javax.cache.event.CacheEntryCreatedListener;
+import javax.cache.event.CacheEntryEvent;
+import javax.cache.event.CacheEntryExpiredListener;
+import javax.cache.event.CacheEntryListenerException;
+import javax.cache.event.CacheEntryRemovedListener;
+import javax.cache.event.CacheEntryUpdatedListener;
+import javax.cache.event.EventType;
+
+import org.testng.Assert;
+
+/**
+ * Borrowed from TCK tests
+ *
+ * @author gustavonalle
+ */
+public class TrackingCacheEntryListener<K, V> implements CacheEntryCreatedListener<K, V>, CacheEntryUpdatedListener<K, V>, CacheEntryExpiredListener<K, V>, CacheEntryRemovedListener<K, V>, Serializable, AutoCloseable {
+   AtomicInteger created = new AtomicInteger();
+   AtomicInteger updated = new AtomicInteger();
+   AtomicInteger removed = new AtomicInteger();
+   ArrayList<CacheEntryEvent<K, V>> entries = new ArrayList<>();
+
+   public TrackingCacheEntryListener() {
+   }
+
+   public int getCreated() {
+      return this.created.get();
+   }
+
+   public int getUpdated() {
+      return this.updated.get();
+   }
+
+   public int getRemoved() {
+      return this.removed.get();
+   }
+
+   public ArrayList<CacheEntryEvent<K, V>> getEntries() {
+      return this.entries;
+   }
+
+   public void onCreated(Iterable<CacheEntryEvent<? extends K, ? extends V>> events) throws CacheEntryListenerException {
+
+      for (Object event1 : events) {
```

Comment:
This could be typed as CacheEntryEvent<? extends K, ? extends V> and the cast wouldn't be needed on the next line.